api4/file_test.go

```diff
@@ -609,10 +621,16 @@ func TestUploadFiles(t *testing.T) {
 							require.Nil(t, err)
 							if !bytes.Equal(data, expected) {
 								tf, err := ioutil.TempFile("", fmt.Sprintf("test_%v_*_%s", i, name))
+								defer tf.Close()
+								require.Nil(t, err)
+								wBytes, err := io.Copy(tf, bytes.NewReader(data))
 								require.Nil(t, err)
-								_, _ = io.Copy(tf, bytes.NewReader(data))
-								tf.Close()
-								t.Errorf("Actual data mismatched %s, written to %q", name, tf.Name())
+								require.Equal(t, int64(len(data)), wBytes)
+								if strings.Contains(name, "test_expected_tiff") {
```

Comment:
Let's add an additional comment on top "// TODO: remove this once MM-22056 is fixed."